### ISSUE TITLE
`tail_expr_drop_order` lint adds too much noise to rustfix

### ISSUE THREAD

**Issue by dev-ardi** [rust-lang/rust#132323]
Lots of local bindings have a significant drop implementation, and the warning is not particularly small.
This completely hides every single other warning. Adding this much noise makes it near impossible to notice any potentially meaningful changes. 

cc: https://github.com/rust-lang/rust/pull/128662

@rustbot label +A-diagnostics +A-edition-2024

**Comment by traviscross** [comment]
@dev-ardi: Perhaps it would help if you could give some examples you've experienced here.

Also, keep in mind that these lints are mostly mean to be applied automatically with `cargo fix --edition`.  Perhaps you could tell us about what you're doing here that's causing you to experience the lint in this way.

cc @dingxiangfei2009

**Comment by dev-ardi** [comment]
When running `cargo fix --edition` on any large repository the stdout gets filled with that.
I tried on my own closed source repo but in any other happens too. Try https://github.com/mozilla/uniffi-rs for example. It's the only warning that gets emited, I think (but I don't know)
Small ones like https://github.com/BurntSushi/ripgrep also have a few but not so many

Also I've noticed that the warning also gets emited randomly without `cargo fix --edition` too...?<
```
warning: these values and local bindings have significant drop implementation that will have a different drop order from that of Edition 2021
   --> redacted.rs:885:44
    |
885 |                     .and_then(|attr| match attr.parse_args::<ViaIntermediary>() {
    |                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = warning: this changes meaning in Rust 2024
    = note: for more information, see issue #123739 <https://github.com/rust-lang/rust/issues/123739>
    = note: `--force-warn tail-expr-drop-order` implied by `--force-warn rust-2024-compatibility`
~``

**Comment by traviscross** [comment]
We believe this is addressed by #131326, as it significantly cut down the noise.

There's still of course some residual noise here.  If you see common patterns to this noise that we might be able to use to further improve the heuristic, please of course file new issues for those.
